src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from 'grid-emotion';
+import SelectControl from 'app/components/forms/selectControl';
+
+import {getInternal, getExternal} from './utils';
+import {TOPK_COUNTS} from '../data';
+
+export default class Aggregation extends React.Component {
+  static propTypes = {
+    value: PropTypes.array,
+    onChange: PropTypes.func,
+    columns: PropTypes.array,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedFunction: null,
+    };
+  }
+
+  getOptions() {
+    const currentValue = getInternal(this.props.value);
+    return [{label: currentValue, value: currentValue}];
+  }
+
+  filterOptions = (options, input, value) => {
+    input = input || this.state.selectedFunction || '';
+
+    let optionList = [
+      {value: 'count', label: 'count'},
+      {value: 'uniq', label: 'uniq(...)'},
+      {value: 'topK', label: 'topK(...)'},
+      {value: 'avg', label: 'avg(...)'},
+    ];
+
+    if (input.startsWith('uniq')) {
+      optionList = this.props.columns.map(({name}) => ({
+        value: `uniq(${name})`,
+        label: `uniq(${name})`,
+      }));
+    }
+
+    if (input.startsWith('avg')) {
+      optionList = this.props.columns
+        .filter(({type}) => type === 'number')
+        .map(({name}) => ({
+          value: `avg(${name})`,
+          label: `avg(${name})`,
+        }));
+    }
+
+    if (input.startsWith('topK')) {
+      optionList = TOPK_COUNTS.map(num => ({
+        value: `topK(${num})`,
+        label: `topK(${num})(...)`,
+      }));
+    }
+
+    const topKValueMatch = input.match(/^topK\((\d+)\)/);
+
+    if (topKValueMatch) {
+      const count = topKValueMatch[1];
+
+      optionList = this.props.columns.map(({name}) => ({
+        value: `topK(${count})(${name})`,
+        label: `topK(${count})(${name})`,
+      }));
+    }
+
+    return optionList.filter(({label}) => label.includes(input));
+  };
+
+  focus() {
+    this.select.focus();
+  }
+
+  handleChange = option => {
```

Comment:
Let's add assertions for component state for the different branches below, and also a test for when `onChange` gets called.